tests/e2e/basic-ingest-redis-serving.py

```diff
@@ -63,101 +63,164 @@ def client(core_url, serving_url, allow_dirty):
-            "FeatureSet correctly or the client needs to wait longer for FeatureSet "
-            "to be committed."
-        )
-
+    # reset client's project to default for other tests
+    client.set_project(DEFAULT_PROJECT)
 
 @pytest.mark.timeout(300)
 @pytest.mark.run(order=11)
-def test_basic_ingest_success(client, basic_dataframe):
-    client.set_project(PROJECT_NAME)
-
+def test_basic_ingest_success(client, cust_trans_df, driver_df):
     cust_trans_fs = client.get_feature_set(name="customer_transactions")
+    driver_fs = client.get_feature_set(name="driver")
 
     # Ingest customer transaction data
-    client.ingest(cust_trans_fs, basic_dataframe)
+    client.ingest(cust_trans_fs, cust_trans_df)
+    client.ingest(driver_fs, driver_df)
     time.sleep(5)
 
 
 @pytest.mark.timeout(45)
 @pytest.mark.run(order=12)
-def test_basic_retrieve_online_success(client, basic_dataframe):
+def test_basic_retrieve_online_success(client, cust_trans_df):
     # Poll serving for feature values until the correct values are returned
     while True:
         time.sleep(1)
-
-        client.set_project(PROJECT_NAME)
-
         response = client.get_online_features(
             entity_rows=[
                 GetOnlineFeaturesRequest.EntityRow(
                     fields={
                         "customer_id": Value(
-                            int64_val=basic_dataframe.iloc[0]["customer_id"]
+                            int64_val=cust_trans_df.iloc[0]["customer_id"]
                         )
                     }
                 )
             ],
+            # Test retrieve with different variations of the string feature refs
             feature_refs=[
                 "daily_transactions",
                 "total_transactions",
-            ],
+            ]
         )  # type: GetOnlineFeaturesResponse
 
         if response is None:
             continue
 
         returned_daily_transactions = float(
             response.field_values[0]
-                .fields[PROJECT_NAME + "/daily_transactions"]
-                .float_val
+            .fields["daily_transactions"]
+            .float_val
         )
         sent_daily_transactions = float(
-            basic_dataframe.iloc[0]["daily_transactions"])
+            cust_trans_df.iloc[0]["daily_transactions"])
 
         if math.isclose(
-                sent_daily_transactions,
-                returned_daily_transactions,
-                abs_tol=FLOAT_TOLERANCE,
+            sent_daily_transactions,
+            returned_daily_transactions,
+            abs_tol=FLOAT_TOLERANCE,
         ):
             break
 
 
+@pytest.mark.timeout(45)
+@pytest.mark.run(order=13)
+def test_basic_retrieve_online_multiple_featureset(client, cust_trans_df, driver_df):
+    # Poll serving for feature values until the correct values are returned
+    while True:
+        time.sleep(1)
+        feature_refs = [
+            "customer_transactions:daily_transactions",
+            "driver:rating",
```

Comment:
Should we maybe omit one feature set here?

Reply:
Corrected to test Feature Set inference.